src/XrdSys/XrdSysE2T.cc

```diff
@@ -55,7 +58,15 @@ int initErrTable()
 // Premap all known error codes.
 //
    for(int i = 1; i <errSlots; i++)
-      {eTxt = strerror(ERRNOBASE + i);
+      {// NOTE: On systems without EAUTH (authentication error; currently all systems but GNU Hurd),
+       // EAUTH is remapped to EBADE ('invalid exchange').  Given there's no current XRootD use of a
+       // syscall that can return EBADE, we assume EBADE really means authentication denied.
+#if defined(EBADE)
+       if (i == EBADE)
+          {eTxt = authenticationFailed;
+       } else
+#endif
```

Comment:
Considering that no one uses GNU Hurd, it seems to me that these `#ifdefs` are not desirable. Also, rather than use an `if` inside a loop, it's better to just overwrite the message for `EBADE` after the `for` loop in my opinion, with something like:

```cpp
e2sMap[EBADE] = "Authentication failed";
```

Reply:
In any case, remapping the error message is just a cosmetic change. I'll wait for @abh3's opinion before merging anything.

Reply:
Hoisted the code, also simplified the patch itself.